test/unit/math/opencl/rev/diag_post_multiply_test.cpp

```diff
@@ -0,0 +1,54 @@
+#ifdef STAN_OPENCL
+#include <stan/math.hpp>
+#include <test/unit/math/opencl/util.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+
+auto diag_post_multiply_functor = [](const auto& a, const auto& b) {
+  return stan::math::diag_post_multiply(a, b);
+};
+
+TEST(OpenCL_diag_post_multiply, prim_rev_values_small) {
+  int N = 2;
+  int M = 3;
+
+  Eigen::MatrixXd a(N, M);
+  a << 1, 2, 3, 4, 5, 6;
+  Eigen::VectorXd b(M);
```

Comment:
If you intend to support row vectors add tests for them.

Reply:
Not intended right now.

Reply:
In that case fix the doxygen for diag_*_multiply, to not mention row vectors.

Reply:
Nevermind, we need to support both, I misunderstood. CPU version supports both and so should this one. Or I will just scratch it for now.